### PR TITLE
Fixed valgrind reported leakage of end_historic_sample at shutdown

### PR DESCRIPTION
**Pull request by jwillemsen**
See #640

### CODE REVIEW

**Comment by mitza-oci** [comment]
Does this need to be a separate PR?  It's already merged in to #641.

**Comment by jwillemsen** [comment]
These are changes for an isolated leak which I merged into #641 to continue testing. Please review and merge these separately.

File: dds/DCPS/RTPS/Sedp.cpp [hunk 1]
```diff
@@ -1584,15 +1584,20 @@ void
 Sedp::Writer::end_historic_samples(const DCPS::RepoId& reader)
 {
   const void* pReader = static_cast<const void*>(&reader);
-  ACE_Message_Block mb(DCPS::DataSampleHeader::max_marshaled_size(),
-                       ACE_Message_Block::MB_DATA,
-                       new ACE_Message_Block(static_cast<const char*>(pReader),
-                                             sizeof(reader)));
-  mb.set_flags(ACE_Message_Block::DONT_DELETE);
```
Comment (mitza-oci) [inline dds/DCPS/RTPS/Sedp.cpp]: Since DONT_DELETE is no longer used, update the methods control_delivered() and control_dropped() to simplify them.

Comment (jwillemsen) [inline dds/DCPS/RTPS/Sedp.cpp]: Done

**Pull request merged**